pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6035,3 +6035,194 @@ func createKafkaJbpmObject(dateFormat string, tasksTopics string, casesTopics st
+	return &api.CORSFiltersObject{
+		Filters:               "AC_ALLOW_ORIGIN,AC_ALLOW_METHODS,AC_ALLOW_HEADERS,AC_ALLOW_CREDENTIALS,AC_MAX_AGE",
+		AllowOriginName:       "Access-Control-Allow-Origin",
+		AllowOriginValue:      "*",
+		AllowMethodsName:      "Access-Control-Allow-Methods",
+		AllowMethodsValue:     "GET, POST, OPTIONS, PUT",
+		AllowHeadersName:      "Access-Control-Allow-Headers",
+		AllowHeadersValue:     "Accept, Authorization, Content-Type, X-Requested-With",
+		AllowCredentialsName:  "Access-Control-Allow-Credentials",
+		AllowCredentialsValue: Pbool(true),
+		MaxAgeName:            "Access-Control-Max-Age",
+		MaxAgeValue:           Pint32(1),
+	}
+}
+
+func checkCors(t *testing.T, cors *api.CORSFiltersObject) {
+	assert.Equal(t, cors.Filters, "AC_ALLOW_ORIGIN,AC_ALLOW_METHODS,AC_ALLOW_HEADERS,AC_ALLOW_CREDENTIALS,AC_MAX_AGE")
+	assert.Equal(t, cors.AllowOriginName, "Access-Control-Allow-Origin")
+	assert.Equal(t, cors.AllowOriginValue, "*")
+	assert.Equal(t, cors.AllowMethodsName, "Access-Control-Allow-Methods")
+	assert.Equal(t, cors.AllowMethodsValue, "GET, POST, OPTIONS, PUT")
+	assert.Equal(t, cors.AllowHeadersName, "Access-Control-Allow-Headers")
+	assert.Equal(t, cors.AllowHeadersValue, "Accept, Authorization, Content-Type, X-Requested-With")
+	assert.Equal(t, cors.AllowCredentialsName, "Access-Control-Allow-Credentials")
+	assert.True(t, *cors.AllowCredentialsValue)
+	assert.Equal(t, cors.MaxAgeName, "Access-Control-Max-Age")
+	assert.Equal(t, cors.MaxAgeValue, Pint32(1))
+}
+
+func checkConsoleCORSAssertions(t *testing.T, cr *api.KieApp, env api.Environment) {
+	corsEnabled := isCORSEnabled(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env)
+	assert.True(t, corsEnabled)
+	assert.NotNil(t, cr.Spec.Objects.Console)
+	cors := cr.Status.Applied.Objects.Console.Cors
+	assert.NotNil(t, cors)
+	checkCors(t, cors)
+}
+
+func checkKieServerCORSAssertions(t *testing.T, cr *api.KieApp, env api.Environment) {
+	corsEnabled := isCORSEnabled(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env)
+	assert.True(t, corsEnabled)
+	assert.Len(t, cr.Spec.Objects.Servers, 1)
+	cors := cr.Status.Applied.Objects.Servers[0].Cors
+	assert.NotNil(t, cors)
+	checkCors(t, cors)
+}
+
+func checkDashbuilderCORSAssertions(t *testing.T, cr *api.KieApp, env api.Environment) {
+	corsEnabled := isCORSEnabled(env.Dashbuilder.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env)
+	assert.True(t, corsEnabled)
+	assert.NotNil(t, cr.Spec.Objects.Dashbuilder)
+	cors := cr.Status.Applied.Objects.Dashbuilder.Cors
+	assert.NotNil(t, cors)
+	checkCors(t, cors)
+}
+
+func isCORSEnabled(envs []corev1.EnvVar) bool {
+	corsEnabled := false
+	for _, env := range envs {
+		if strings.HasPrefix(env.Name, "AC_ALLOW") || strings.HasPrefix(env.Name, "FILTERS") {
+			corsEnabled, _ = strconv.ParseBool(env.Value)
+		}
+	}
+	return corsEnabled
+}
+
+func TestEnvironmentWithoutCORS(t *testing.T) {
```

Comment:
can you please also add a test to check something like setting only one filter?

e.g:

```
api.CORSFiltersObject{
		Filters:               "AC_ALLOW_ORIGIN,AC_ALLOW_METHODS,AC_ALLOW_HEADERS,AC_ALLOW_CREDENTIALS,AC_MAX_AGE",
		AllowOriginName:       "Access-Control-Allow-Origin-custom-test",
		AllowOriginValue:      "custom-test-value"

```

and check if any other filter is added on the deployment config, and also on the running bc/kieserver/dashbuilder

Reply:
Ok